### PR TITLE
Remediate code scanning alerts

### PR DESCRIPTION
**Pull request by bvu12**
We have some code scanning alerts in this repo: https://github.com/procurify/sticky/security/code-scanning, mainly related to using regex to sanitize `html` tags.

* [Github](https://codeql.github.com/codeql-query-help/javascript/js-incomplete-multi-character-sanitization/) recommends using a library to perform sanitization, [rails-html-sanitizer](https://github.com/rails/rails-html-sanitizer) seems to be a well-maintained open-source repo
* I've removed an unused `_search.js` file
* I've bumped up the version of `jQuery` to `3.6.3`

I've tested locally and everything seems to work alright 🙂:
image

### CODE REVIEW

File: api_docs/source/javascripts/app/_toc.js [hunk 1]
```diff
@@ -82,7 +81,7 @@
         }
         var thisTitle = $best.data("title");
         if (thisTitle !== undefined && thisTitle.length > 0) {
-          document.title = thisTitle.replace(htmlPattern, "") + " – " + originalTitle;
+          document.title = thisTitle + " – " + originalTitle;  // E.g. 'Create Account POST - API Documentation'
```
Comment (bvu12) [inline api_docs/source/javascripts/app/_toc.js:84]: Based on the endpoint the user is currently looking at, the title of the browser gets updated to `Endpoint - API Documentation`. As we don't do anything fancy with these titles, there's no need for regex sanitization.

File: api_docs/source/javascripts/app/_search.js [hunk 2]
```diff
@@ -1,102 +0,0 @@
-//= require ../lib/_lunr
```
Comment (bvu12) [inline api_docs/source/javascripts/app/_search.js:1]: This functionality is unused - all the searching can be performed through the browser (`CTRL + F`) as everything is on a single page: https://procurify.github.io/sticky/#netsuite.

File: api_docs/Gemfile [hunk 3]
```diff
@@ -11,3 +11,4 @@ gem 'redcarpet', '~> 3.5.0'
 gem 'nokogiri', '~> 1.13.10'
 gem 'sass'
 gem 'webrick'
+gem 'rails-html-sanitizer'
```
Comment (bvu12) [inline api_docs/Gemfile:14]: `Gemfile` is equivalent to `package.json` or `requirements.txt`

File: api_docs/source/javascripts/lib/_jquery.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*!
- * jQuery JavaScript Library v3.6.0
+ * jQuery JavaScript Library v3.6.3
```
Comment (bvu12) [inline api_docs/source/javascripts/lib/_jquery.js:2]: Just bumping the `jQuery` version to `3.6.3`

**Review by gondek — APPROVED** [review]
It's always fun to be reminded of Ruby syntax 😝

**Review by adilkhokhar — APPROVED** [review]
Impressive :)

**Pull request merged**